src/answers-umd.js

```diff
@@ -167,32 +166,45 @@ class Answers {
     parsedConfig.search = new SearchConfig(parsedConfig.search);
     parsedConfig.verticalPages = new VerticalPagesConfig(parsedConfig.verticalPages);
 
-    const storage = new Storage().init(window.location.search);
-    const persistentStorage = new PersistentStorage({
-      updateListener: parsedConfig.onStateChange,
-      resetListener: data => {
-        if (!data[StorageKeys.QUERY]) {
+    const storage = new Storage({
+      update: (data, url) => {
+        if (parsedConfig.onStateChange) {
+          parsedConfig.onStateChange(Object.fromEntries(data), url);
+        }
+      },
+      /**
+       * Called when the 'popstate' event is fired, i.e. window.history
```

Comment:
Love the jsdoc, how do we feel about putting it in update as well?

Reply:
hmm it seems kind of weird to put it twice, I'll move it into the storage.js file